Plugin/CppApi/source/ArcGISArView.cpp

```diff
@@ -0,0 +1,309 @@
+/*******************************************************************************
+ *  Copyright 2012-2019 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#include "ArcGISArView.h"
+#include "TransformationMatrix.h"
+#include "TransformationMatrixCameraController.h"
+#include <QQuickWindow>
+#include <QScreen>
+
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::ArcGISArView
+  \ingroup AR
+  \inmodule ArcGISQtToolkit
+  \since Esri::ArcGISRuntime 100.6
+  \brief Render and tracks the camera.
+ */
+
+using namespace Esri::ArcGISRuntime;
+using namespace Esri::ArcGISRuntime::Toolkit;
+using namespace Esri::ArcGISRuntime::Toolkit::Internal;
+
+/*!
+  \brief A constructor that accepts an optional \a parent.
+ */
+ArcGISArView::ArcGISArView(QQuickItem* parent):
+  ArcGISArViewInterface(parent),
+  m_tmcc(new TransformationMatrixCameraController(this)),
+  m_initialTransformation(TransformationMatrix::createIdentityMatrix(this))
+{
+  connect(m_tmcc, &TransformationMatrixCameraController::originCameraChanged, this, &ArcGISArView::originCameraChanged);
+}
+
+/*!
+  \brief A constructor that accepts an optional \a parent.
```

Comment:
@GuillaumeBelz  Guilluame - This constructor has unimplemented parameters, should this be marked as "internal" for now?